homeassistant/components/palazzetti/climate.py

```diff
@@ -0,0 +1,159 @@
+
+    @property
+    def supported_features(self) -> ClimateEntityFeature:
+        """Supported features."""
+        features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+        )
+        if self.coordinator.client.has_on_off_switch:
+            features |= ClimateEntityFeature.TURN_ON | ClimateEntityFeature.TURN_OFF
+        return features
+
+    @property
+    def hvac_modes(self) -> list[HVACMode]:
+        """List the hvac modes.
+
+        The on/off switch becomes unavaiables at times, during ignition and shut off cycles.
+        An API call to turn on/off during that time will not raise an error but it will fail.
+        """
+        if self.coordinator.client.has_on_off_switch:
+            return [HVACMode.OFF, HVACMode.HEAT]
+        return [self.hvac_mode]
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat or off mode."""
+        is_heating = bool(self.coordinator.client.is_heating)
+        return HVACMode.HEAT if is_heating else HVACMode.OFF
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        try:
+            await self.coordinator.client.set_on(hvac_mode != HVACMode.OFF)
+        except (CommunicationError, ValidationError) as err:
+            raise ServiceValidationError(
+                err, translation_domain=DOMAIN, translation_key=ACTION_NOT_UNAVAILABLE
+            ) from err
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return current temperature."""
+        api_state = self.coordinator.client.room_temperature
+        return api_state if isinstance(api_state, float) else None
+
+    @property
+    def target_temperature(self) -> int | None:
+        """Return the temperature."""
+        api_state = self.coordinator.client.target_temperature
+        return api_state if isinstance(api_state, int) else None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+        temperature = cast(float, kwargs.get(ATTR_TEMPERATURE))
+        await self.coordinator.client.set_target_temperature(int(temperature))
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the fan mode."""
+        api_state = self.coordinator.client.fan_speed
+        return FAN_MODES[api_state]
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set new fan mode."""
+
+        try:
+            if fan_mode == FAN_SILENT:
+                await self.coordinator.client.set_fan_silent()
+            elif fan_mode == FAN_HIGH:
+                await self.coordinator.client.set_fan_high()
+            elif fan_mode == FAN_AUTO:
+                await self.coordinator.client.set_fan_auto()
+            else:
+                await self.coordinator.client.set_fan_speed(FAN_MODES.index(fan_mode))
+        except (CommunicationError, ValidationError) as err:
```

Comment:
What's a validation error?

Reply:
But is that something the user should act on or something temporary? In that case we should raise ServiceValidationError